### PR TITLE
feat: add CSV export of epoch transactions for address

### LINKED ISSUE #11194: Celo: CSV export of epoch transactions for address

### LINKED ISSUE #11201: Pagination in election rewards for delegated payments
Doesn't work 

https://celo-alfajores.blockscout.com/api/v2/addresses/0x456f41406B32c45D59E539e4BBA3D7898c3584dA/election-rewards

### PR DESCRIPTION
**Pull request by fedor-ivn**
Closes #11194, closes #11201.

## Motivation

The Celo community requested that validators retain the ability to export epoch transaction data when migrating to our instance. This feature is crucial for calculating earnings from participating in validator groups within the Celo protocol.

For context, the legacy Celo explorer provides a similar feature on the [validator address page](https://explorer.celo.org/mainnet/address/0xc24baeac0Fd189637112B7e33d22FfF2730aF993/epoch-transactions), where users can generate a [CSV export](https://explorer.celo.org/mainnet/csv-export?address=0xc24baeac0Fd189637112B7e33d22FfF2730aF993&type=epoch-transactions) for this data.

### Enhancements

- Introduced a new `/api/v1/celo-election-rewards-csv` endpoint and the `AddressElectionRewardsCsvExporter` module.

**⚠️ Note**: The exported table does not preserve the "LockedGold" and "VotingGold" fields, since this data is not indexed.

### Bug fixes

- Pagination of election rewards when the type of reward is `delegated_payment`

## Checklist for your Pull Request (PR)

- [ ] If I added new functionality, I added tests covering it.
- [ ] If I fixed a bug, I added a regression test to prevent the bug from silently reappearing again.
- [ ] I checked whether I should update the docs and did so by submitting a PR to [docs repository](https://github.com/blockscout/docs).
- [ ] If I added/changed/removed ENV var, I submitted a PR to [docs repository](https://github.com/blockscout/docs) to update the list of [env vars](https://github.com/blockscout/docs/blob/master/for-developers/information-and-settings/env-variables.md) and I updated the version to `master` in the Version column. If I removed variable, I added it to [Deprecated ENV Variables](https://github.com/blockscout/docs/blob/master/for-developers/information-and-settings/env-variables/deprecated-env-variables/README.md) page. After merging docs PR, changes will be reflected in these [pages](https://docs.blockscout.com/for-developers/information-and-settings/env-variables).
- [ ] If I added new DB indices, I checked, that they are not redundant, with PGHero or other tools.
- [ ] If I added/removed chain type, I modified the Github CI matrix and PR labels accordingly.

### CODE REVIEW

File: apps/explorer/lib/explorer/chain/csv_export/celo/address_election_rewards_csv_exporter.ex [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+          # ValidatorGroupAddress
+          (reward.type in ~w(validator voter) && reward.associated_account_address_hash) || "N/A",
+          # ToAddress
+          reward.account_address_hash,
+          # Type
+          "IN",
+          # Value
+          reward.amount |> Wei.to(:ether) |> Decimal.to_string(:normal),
+          # ValueInWei
+          reward.amount |> Wei.to(:wei) |> Decimal.to_string(:normal),
```
Comment (fedor-ivn) [inline apps/explorer/lib/explorer/chain/csv_export/celo/address_election_rewards_csv_exporter.ex:77]: Should we avoid scientific notation here like `1E-27`?

File: apps/explorer/lib/explorer/chain/celo/election_reward.ex [hunk 2]
```diff
@@ -341,6 +335,7 @@ defmodule Explorer.Chain.Celo.ElectionReward do
     end
   end
 
+  # sobelow_skip ["DOS.StringToAtom"]
```
Comment (sl1depengwyn) [inline apps/explorer/lib/explorer/chain/celo/election_reward.ex:338]: This is actually a DOS possibility, lets define `@string_types_enum` or do it in some other way than calling `String.to_atom`

File: apps/explorer/lib/explorer/chain/csv_export/celo/address_election_rewards_csv_exporter.ex (hunk 1 above)
Comment (sl1depengwyn) [reply apps/explorer/lib/explorer/chain/csv_export/celo/address_election_rewards_csv_exporter.ex:77]: I would avoid it

File: apps/explorer/lib/explorer/chain/celo/election_reward.ex [hunk 3]
```diff
@@ -521,4 +516,32 @@ defmodule Explorer.Chain.Celo.ElectionReward do
...
   end
+
+  @doc """
+  Custom filter for `ElectionReward`, inspired by
+  `Chain.where_block_number_in_period/3`.
+
+  TODO: Consider reusing `Chain.where_block_number_in_period/3`. This would
+  require storing or making `merge_select` of `block_number`.
+  """
+  def where_block_number_in_period(base_query, from_block, to_block)
```
Comment (sl1depengwyn) [inline apps/explorer/lib/explorer/chain/celo/election_reward.ex:527]:
```suggestion
  @spec where_block_number_in_period(Ecto.Query.t(), String.t() | integer() | nil, String.t() | integer() | nil) ::
          Ecto.Query.t()
  def where_block_number_in_period(base_query, from_block, to_block)
```

**Review by sl1depengwyn — CHANGES_REQUESTED** [review]

**Review by sl1depengwyn — APPROVED** [review]

File: apps/explorer/lib/explorer/chain/celo/reader.ex [hunk 4]
```diff
@@ -8,18 +8,17 @@ defmodule Explorer.Chain.Celo.Reader do
   import Explorer.Chain,
     only: [
       select_repo: 1,
-      join_associations: 2,
-      default_paging_options: 0,
-      max_consensus_block_number: 1
+      join_associations: 2
     ]
 
+  alias Explorer.Chain
```
[the remark this answers is not in the document]
Comment (fedor-ivn) [reply apps/explorer/lib/explorer/chain/celo/reader.ex]: There's no rationale I guess, probably introduced by accident. Fixed ✅

**Review by vbaranov — APPROVED** [review]

**Pull request merged**